plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -1973,15 +1973,18 @@ regexSingleMatch msg regex = case matchRegexUnifySpaces msg regex of
     _          -> Nothing
 
 -- | Process a list of (module_name, filename:src_span) values
--- | Eg. [(Data.Map, app/ModuleB.hs:2:1-18), (Data.HashMap.Strict, app/ModuleB.hs:3:1-29)]
+--
+-- Eg. [(Data.Map, app/ModuleB.hs:2:1-18), (Data.HashMap.Strict, app/ModuleB.hs:3:1-29)]
 regExImports :: T.Text -> Maybe [(T.Text, T.Text)]
 regExImports msg
     | Just mods' <- allMatchRegex msg "‘([^’]*)’"
     , Just srcspans' <- allMatchRegex msg
+    -- This regex has to be able to deal both with single-line srcpans like "(/path/to/File.hs:2:1-18)"
+    -- as well as multi-line srcspans like "(/path/to/File.hs:(3,1)-(5,2))"
 #if MIN_VERSION_ghc(9,7,0)
-                          "\\(at ([^)]*)\\)"
+                          "\\(at ([^:]+:[^ ]+)\\)"
 #else
-                          "\\(([^)]*)\\)"
+                          "\\(([^:]+:[^ ]+)\\)"
```

Comment:
Since this is almost the same, do you think it is a good idea to remove the CPP and use a regex with an optional group like:
`"\\((at )?([^:]+:[^ ]+)\\)"`

Reply:
Unfortunately this doesn't work, because it introduces a new capturing group in the regex, which changes the number of matches and because of that it fails the check on this line:

https://github.com/haskell/haskell-language-server/blob/97aac543bc0c80dc3c1760f716653789859a9535/plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs#L1989

Reply:
I'll wait for further suggestions if you have any and will merge it in the afternoon otherwise.